auth/src/main/java/com/firebase/ui/auth/provider/TwitterProvider.java

```diff
@@ -60,8 +60,18 @@ public void startLogin(Activity activity) {
     }
 
     @Override
-    public void success(Result<TwitterSession> result) {
-        mCallbackObject.onSuccess(createIdpResponse(result.data));
+    public void success(final Result<TwitterSession> sessionResult) {
+        mTwitterAuthClient.requestEmail(sessionResult.data, new Callback<String>() {
+            @Override
+            public void success(Result<String> result) {
+                mCallbackObject.onSuccess(createIdpResponse(sessionResult.data, result.data));
+            }
+
+            @Override
+            public void failure(TwitterException exception) {
+                TwitterProvider.this.failure(exception);
```

Comment:
This is a problem:

> Even if the user grants access to her email address, it is not guaranteed you will get an email address. For example, if someone signed up for Twitter with a phone number instead of an email address, the email field may be empty. When this happens, the failure method will be called because there is no email address available.

It will be confusing to the user if they press authorize and the sign in just fails unexpectedly because they didn't give Twitter an email. The only place we need to display a message is in `AuthMethodPickerActivity`, but what should we say and how? A snackbar? An alert dialog?